src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs

```diff
@@ -506,19 +506,22 @@ private bool ShouldBeInStateSyncMode(Snapshot best)
         private bool CanBeInSnapRangesPhase(Snapshot best)
         {
             bool isCloseToHead = best.PeerBlock >= best.Header && (best.PeerBlock - best.Header) < Constants.MaxDistanceFromHead;
+            bool isFullStateAncientOrNotFound = best.PeerBlock - best.State >= FastSyncCatchUpHeightDelta || best.State == 0;
```

Comment:
Isn't this check inherited from `ShouldBeInStateSyncMode` more precisely from here: https://github.com/NethermindEth/nethermind/blob/20dfead304eb9eb9bbff6f4351f58d00f86579d2/src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs#L477
which is based on:
https://github.com/NethermindEth/nethermind/blob/20dfead304eb9eb9bbff6f4351f58d00f86579d2/src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs#L283-L289

then this check looks redundant and not needed.